docs/proposal-architecture.md

```diff
@@ -0,0 +1,111 @@
+# Architecture Proposal
+
+## Problem Setting
+
+- Data
+    - get data from http://www.ordinal.com/gensort.html
+        - one record is consist of 10 bytes of key and 90 bytes of value 
+        - sort data using key with lexicographical order
+- Resources
+    - one Master Node
+    - fixed number of Worker Nodes 
+- Input & Output
+    - Input: bunch of 32MB files which is allocated in each Worker
+    - Output: ordering of ???MB files with sorted data, in each Worker 
+
+## Goal
+- Implement system that can sort key/value records stored across **multiple disks**, using **multiple machines** with **distributed/parallel** programming
+
+## Method
+
+Basically we are going to use merge sort. To fully exploit N number of Workers, we have to sort parallel in each Worker. So hole procedure could be divided into two phase as below.
+
+1. Relocate data in each Worker.
+2. Each Worker sort its relocated data in each Worker's disk.
+
+Now it's time to deep dive into each two phase with specific step.
+
+### First Phase 
+To exploit all the Workers in parallel, we have to distribute the recordes into each Worker.
+
+#### Step1: Data Sampling
+- Sample a small subset of records from each Worker Node.
```

Comment:
I think it would be better to explain the purpose of the sampling and provide detailed information on how it's done

Reply:
Got it